docs/configuration.md

```diff
@@ -376,7 +376,15 @@ are included below.
 Make sure to check the READMEs of each of these packages, which explain how to configure the
 various parts of the Emacs integration.
 In particular, `lsp-haskell` provides customization options for the `haskell-language-server`-specific parts,
-such as the path to the server executable.
+such as the path to the server executable, or configuring the tactics plugin. For example:
```

Comment:
This is in conflict with the previous clause, which says that `lsp-haskell` provides customization options... and then you give an example of how to bypass them! I still really think this should be in the `lsp-haskell` readme if anywhere. Better to keep the Emacs docs with the Emacs package.

Reply:
FYI, michael means this repository: https://github.com/emacs-lsp/lsp-haskell/

Reply:
@michaelpj I see what you're saying and personally agree that it should be in `lsp-haskell` (though, in addition to being in HLS's docs). If there's a way to configure `lsp-haskell` better than `lsp--set-configuration`, I think that would be relevant to have in HLS's docs instead of what I suggested, but, `lsp-haskell` doesn't have any documentation I could find suggesting just _what_ incantation it wants to configure variables, like, tactics configuration.

Reply:
> I see now that lsp-haskell does mention (customize-group 'lsp-haskell), but that doesn't have the tactics config in it

Alright, there we go. That's a bug. Which I have been procrastinating on fixing for a while, but you pushed me over the edge ;) https://github.com/emacs-lsp/lsp-haskell/pull/144 .

> Could you suggest a better alternative (IE the proper way to configure lsp-haskell), or point me to documentation?

Could you rather help me figure out what route you took in looking for information so we can make the right route clearer? Evidently you didn't find the bit in the `lsp-haskell` readme that you needed and I'd like to understand why!

The route I was hoping people would take is:
- HLS configuration doc
- `lsp-haskell` readme (since the configuration doc tells you to go there)
- "Emacs configuration" section of `lsp-haskell` readme, which talks about the customization options

> I think that would be relevant to have in HLS's docs instead of (or in addition to) what I suggested.

Perhaps I'm being overly dogmatic, but I would really prefer to keep the Emacs documentation next to the Emacs package, we're much more likely to keep it up-to-date that way!

-----

Also, I would definitely appreciate a PR to the `lsp-haskell` readme with your workaround for configuring settings that aren't exposed. That's a very useful trick for when we haven't got an option for something yet!

Reply:
understood, I'll close this PR then, and offer this one in its place: https://github.com/emacs-lsp/lsp-haskell/pull/146

> what route you took in looking for information 

My expectations were to find these answers first in HLS's README, second in HLS's docs, third in... probably lsp-mode was third, fourth was lsp-haskell, I missed the `customize-group` comment in lsp-haskell's readme (it _was_ in the configuration heading, but a lot of that heading was just installation pointers so I just missed it toward the end, and it wouldn't have helped since it had fallen out of parity with HLS (thanks for your new pr btw!)), so, I probably poked at lsp-ui next, then I hunted through github issues of various projects via search engine until I stumbled across someone configuring a python project using the same technique I show here, and extrapolated that the proper configuration key paths would align with those in the docs, which, still took some fiddling to get right.

I'm super grateful for all you guys have done though, so, massive thanks and sorry I submitted this PR in the wrong place.

Reply:
@freckeltonj Don't worry and thanks again for the patch wherever is finally merged
And for explain your path looking for info, that pov is invaluable for helping improve docs

Reply:
@freckletonj, I know how that is. Docs are indeed frequently hard to merge. Besides Wadler's Law, docs workflow essentially should be built on the Cunningham Wiki principles, but projects frequently use code review workflow for them. (I in no way appeal "to authority" there, its mine conclusions in concise public-available terms).

Reply:
I love the wiki idea. Actually, my favorite all-time documentation is kinda similar, have you ever seen [ClojureDocs](https://clojuredocs.org/)? I seriously think haskell would benefit from this format.

Reply:
Heard good stuff about ClosureDocs.

Essentially made alike Haskell docs for myself to learn the language.

Something like ClosureDocs may start/be tried in near time. Especially after the new base. But HaskellWiki is already there & so as GHC wiki. Back in the day there was an idea/try to migrate Hackage package docs into HaskellWiki, so every package had an article there. As put "a wiki that needs to invent its own HTML for markup is a failed design", which is one of the things why people are not enthusiastic about writing in MediaWikis & HaskellWiki also. & Hoogle is there. So basically seems like if a couple of people would not spearhead the agenda for several years & manage to connect enough pieces together - likely the situation would stay a cluster ***.

Haskell does not have good basic tutorial open learning materials for newcomers.

But overall I consider the situation not ideal, but not that bad either, it is bearable.

---

It is easier to give users the ability to change docs. & sit & review changes done by users in the last week/month (even team-sporadically/chaotically) & improve changes even more, than for 4 people to review 1 paragraph of documentation.

Devs anyway never (to be safe - almost never) change manual docs in the same PR they change code, even because code review changes would change the doc updates.

---

Just to be clear, the power of wikis I mention, is in these:

![Screenshot-2021-12-15-00:36:17](https://user-images.githubusercontent.com/20933385/146090436-edba1746-1b99-4229-ab3a-69055be8403c.png)

That is what I mean by "cornerstone wiki design principles".

All ideas basically converge on having an easily editable community Zettelkasten. MediaWiki (Wikipedia-related wikis) are a bad example due to formatting. All wikis also should be interlinked & preview articles of one another. MediaWikis have those integration abilities.

I've seen several contemporary innovative approaches to docs - they all converge to have community Zettelkastens. Even back in the day, Oxford English dictionary was created so.